cf/data/data.py

```diff
@@ -5414,13 +5673,12 @@ def _Units(self):
     def _cyclic(self):
         """Storage for axis cyclicity.
 
-        Identifies which axes are cyclic (and therefore allow cyclic
-        slicing).
-
-        It must be a subset of the axis identifiers given by the
-        `_axes` attribute.
+        Contains a `set` that identifies which axes are cyclic (and
+        therefore allow cyclic slicing). The set contains a subset of
+        the axis identifiers defined by the `_axes` attribute.
 
-        .. warning:: Never change the `_cyclic` attribute in-place.
+        .. warning:: Never change the value of the `_cyclic` attribute
```

Comment:
Can we forbid `_cyclic` from being changed instead of warning strongly against it? Or is it perhaps not possible to do so due to some facet of laziness or similar? Or just not a good idea?

Reply:
Hmm. Good idea. we could get rid of `@setter._cyclic`, and instead of exposing some nasty internals (i.e. the custom dictionary) we could re-bury them in a setter method: `def _set_cyclic(self, value)`. Let's open another PR for that.

Reply:
Sounds like a good plan to me! Do you want to do that PR, or shall I (I am happy to, but you may already have started and/or want to implement it yourself)?

Reply:
All yours, thanks.